nexus_constructor/transformations.py

```diff
@@ -157,14 +157,29 @@ def ui_value(self) -> float:
         Used for getting the 3d view magnitude (as a placeholder or if the dataset is scalar)
         :return:
         """
-        if isinstance(self.dataset, h5py.Dataset) and np.isscalar(self.dataset[()]):
-            try:
-                float(self._dataset[()])
-                return self._dataset[()]
-            except ValueError:
-                logging.debug(
-                    "transformation value is not cast-able to int, using UI placeholder value instead."
-                )
+        if isinstance(self.dataset, h5py.Dataset):
+            if np.isscalar(self.dataset[()]):
+                try:
+                    self.ui_value = float(self._dataset[()])
+                    return float(self._dataset[()])
+                except ValueError:
+                    logging.debug(
+                        "transformation value is not cast-able to float/int, using UI placeholder value instead."
+                    )
+            else:
+                # dataset value is array
```

Comment:
Would be good to explain in the comment, or ideally a unit test (possible?), that we'll use the first value in the array for 3D view display purposes.

Reply:
Have added to the comment and added a couple of unit tests to cover.